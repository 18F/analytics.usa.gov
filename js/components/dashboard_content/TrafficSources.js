import React from "react";
import PropTypes from "prop-types";

import ConsolidatedBarChart from "../chart/ConsolidatedBarChart";
import Tooltip from "../tooltip/Tooltip";

/**
 * Contains charts and other data visualizations for the traffic sources
 * section of the site. This component is mainly laying out the structure for
 * the section and passes props necessary for getting data and displaying
 * visualizations to child components.
 *
 * @param {object} props the properties for the component
 * @param {string} props.dataHrefBase the URL of the base location of the data
 * to be downloaded including the agency path. In production this is proxied and
 * redirected to the S3 bucket URL.
 * @returns {import('react').ReactElement} The rendered element
 */
function TrafficSources({ dataHrefBase }) {
  return (
    <div className="padding-top-1 grid-row">
      <div
        id="session_channel_groups"
        className="desktop:grid-col-6 desktop:padding-right-2"
      >
        <div className="chart__title">
          <a href="/definitions#dimension_default_channel_group">
            <Tooltip
              position="top"
              content="Channel groupings are rule-based definitions of your traffic sources."
            >
              Top Channels
            </Tooltip>
          </a>
          <a
            href={`${dataHrefBase}/top-session-channel-group-30-days.csv`}
            aria-label="top-session-channel-group-30-days.csv"
          >
            <svg
              className="usa-icon margin-bottom-neg-05 margin-left-05"
              aria-hidden="true"
              focusable="false"
              role="img"
            >
              <use xlinkHref="/assets/uswds/img/sprite.svg#file_present"></use>
            </svg>
          </a>
        </div>
        <div className="text--capitalize">
          <ConsolidatedBarChart
            dataUrl={`${dataHrefBase}/top-session-channel-group-30-days.json`}
            chartDataKey={"session_default_channel_group"}
            maxItems={10}
          />
        </div>
      </div>

      <div
        id="session_source_mediums"
        className="desktop:grid-col-6 desktop:padding-left-2"
      >
        <div className="chart__title">
          <a href="/definitions#dimension_source_medium">
            <Tooltip
              position="top"
              content="A source is a representation of the publisher or inventory source from which traffic originated. A medium is a method for acquiring users to websites or applications."
            >
              Top Sources/Media
            </Tooltip>
          </a>
          <a
            href={`${dataHrefBase}/top-session-source-medium-30-days.csv`}
            aria-label="top-session-source-medium-30-days.csv"
          >
            <svg
              className="usa-icon margin-bottom-neg-05 margin-left-05"
              aria-hidden="true"
              focusable="false"
              role="img"
            >
              <use xlinkHref="/assets/uswds/img/sprite.svg#file_present"></use>
            </svg>
          </a>
        </div>
        <div className="text--capitalize">
          <ConsolidatedBarChart
            dataUrl={`${dataHrefBase}/top-session-source-medium-30-days.json`}
            chartDataKey={"session_source_medium"}
            maxItems={10}
          />
        </div>
      </div>
    </div>
  );
}

TrafficSources.propTypes = {
  dataHrefBase: PropTypes.string.isRequired,
};

export default TrafficSources;
