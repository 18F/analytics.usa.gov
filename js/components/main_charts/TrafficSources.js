import React from "react";
import PropTypes from "prop-types";

import TopChannels from "./TopChannels";
import TopSourceMedia from "./TopSourceMedia";

/**
 * Contains charts and other data visualizations for the traffic sources
 * section of the site. This component is mainly laying out the structure for
 * the section and passes props necessary for getting data and displaying
 * visualizations to child components.
 *
 * @param {object} props the properties for the component
 * @param {string} props.dataHrefBase the URL of the base location of the data
 * to be downloaded including the agency path. In production this is proxied and
 * redirected to the S3 bucket URL.
 * @returns {import('react').ReactElement} The rendered element
 */
function TrafficSources({ dataHrefBase }) {
  return (
    <>
      <section
        id="session_channel_groups"
        className="desktop:grid-col-6 bar-chart-component"
      >
        <h4>Top Channels</h4>
        <TopChannels dataHrefBase={dataHrefBase} />
      </section>

      <section
        id="session_source_mediums"
        className="desktop:grid-col-6 bar-chart-component"
      >
        <h4>Top Sources/Media</h4>
        <TopSourceMedia dataHrefBase={dataHrefBase} />
      </section>
    </>
  );
}

TrafficSources.propTypes = {
  dataHrefBase: PropTypes.string.isRequired,
};

export default TrafficSources;
