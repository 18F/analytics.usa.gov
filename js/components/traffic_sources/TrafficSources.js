import React from "react";
import PropTypes from "prop-types";

import TopChannels from "./TopChannels";
import TopSourceMedia from "./TopSourceMedia";

function TrafficSources({ dataURL, dataPrefix }) {
  const dataHrefBase = `${dataURL}/${dataPrefix}`;

  return (
    <>
      <section
        id="session_channel_groups"
        className="desktop:grid-col-6 bar-chart-component"
      >
        <h4>Top Channels</h4>
        <TopChannels dataHrefBase={dataHrefBase} />
      </section>

      <section
        id="session_source_mediums"
        className="desktop:grid-col-6 bar-chart-component"
      >
        <h4>Top Sources/Media</h4>
        <TopSourceMedia dataHrefBase={dataHrefBase} />
      </section>
    </>
  );
}

TrafficSources.propTypes = {
  dataURL: PropTypes.string.isRequired,
  dataPrefix: PropTypes.string.isRequired,
};

export default TrafficSources;
