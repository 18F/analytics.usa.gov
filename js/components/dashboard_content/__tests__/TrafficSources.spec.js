import React from "react";
import { render, screen, waitFor } from "@testing-library/react";

import TrafficSources from "../TrafficSources";

describe("TrafficSources", () => {
  let component;

  beforeEach(async () => {
    component = render(
      <TrafficSources dataHrefBase="http://www.example.com/data/" />,
    );
    await waitFor(() => screen.getByText("Top Channels"));
  });

  it("renders", () => {
    expect(component.asFragment()).toMatchSnapshot();
  });
});
