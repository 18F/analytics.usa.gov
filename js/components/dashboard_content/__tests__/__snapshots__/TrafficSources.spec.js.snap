// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrafficSources renders 1`] = `
<DocumentFragment>
  <ul
    class="usa-card-group grid-row tablet:grid-gap-2 card:padding-x-1 tablet:padding-x-0"
  >
    <li
      class="usa-card card:grid-col-12 desktop:grid-col-6 card:padding-bottom-2 desktop:padding-bottom-0"
    >
      <div
        class="usa-card__container padding-105 margin-0 border-0 height-auto"
      >
        <div
          class="grid-row"
        >
          <div
            class="chart__title display-flex card:grid-col-12 mobile-lg:grid-col-fill card:flex-justify-center mobile-lg:flex-justify-start card:padding-bottom-105 mobile-lg:padding-bottom-0"
          >
            <a
              aria-label="top-session-channel-group-yesterday.csv"
              href="http://www.example.com/data//top-session-channel-group-yesterday.csv"
            >
              <svg
                aria-hidden="true"
                class="usa-icon margin-bottom-neg-05 margin-left-05"
                focusable="false"
                role="img"
              >
                <use
                  xlink:href="/assets/uswds/img/sprite.svg#file_present"
                />
              </svg>
            </a>
            <a
              href="/definitions#dimension_default_channel_group"
            >
              <span
                class="usa-tooltip"
              >
                <span
                  aria-describedby="tooltip-550000"
                  class="usa-tooltip__trigger"
                  data-position="top"
                  tabindex="0"
                >
                  Top Channels
                </span>
                <span
                  aria-hidden="true"
                  class="usa-tooltip__body"
                  id="tooltip-550000"
                  role="tooltip"
                >
                  Channel groupings are rule-based definitions of your traffic sources.
                </span>
              </span>
            </a>
          </div>
          <div
            class="card:grid-col-12 mobile-lg:grid-col-auto"
          >
            <div
              class="display-flex card:flex-justify-center mobile-lg:flex-justify-end"
            >
              <select
                aria-label="session channel group chart time filter"
                class="usa-select chart__filter margin-top-0"
              >
                <option
                  value="top-session-channel-group-yesterday"
                >
                  Yesterday
                </option>
                <option
                  value="top-session-channel-group-7-days"
                >
                  7 Days
                </option>
                <option
                  value="top-session-channel-group-30-days"
                >
                  30 Days
                </option>
                <option
                  value="top-session-channel-group-90-days"
                >
                  90 Days
                </option>
                <option
                  value="top-session-channel-group-current-year"
                >
                  Current Year
                </option>
                <option
                  value="top-session-channel-group-current-fiscal-year"
                >
                  Current Fiscal Year
                </option>
                <option
                  value="top-session-channel-group-previous-year"
                >
                  Previous Year
                </option>
                <option
                  value="top-session-channel-group-previous-fiscal-year"
                >
                  Previous Fiscal Year
                </option>
              </select>
            </div>
          </div>
        </div>
        <div
          class="text--capitalize"
        >
          <figure
            id="chart_session_default_channel_group"
          >
            <div
              class="data chart__bar-chart margin-top-2"
            />
          </figure>
        </div>
      </div>
    </li>
    <li
      class="usa-card card:grid-col-12 desktop:grid-col-6"
    >
      <div
        class="usa-card__container padding-105 margin-0 border-0 height-auto"
      >
        <div
          class="grid-row"
        >
          <div
            class="chart__title display-flex card:grid-col-12 mobile-lg:grid-col-fill card:flex-justify-center mobile-lg:flex-justify-start card:padding-bottom-105 mobile-lg:padding-bottom-0"
          >
            <a
              aria-label="top-session-source-medium-yesterday.csv"
              href="http://www.example.com/data//top-session-source-medium-yesterday.csv"
            >
              <svg
                aria-hidden="true"
                class="usa-icon margin-bottom-neg-05 margin-left-05"
                focusable="false"
                role="img"
              >
                <use
                  xlink:href="/assets/uswds/img/sprite.svg#file_present"
                />
              </svg>
            </a>
            <a
              href="/definitions#dimension_source_medium"
            >
              <span
                class="usa-tooltip"
              >
                <span
                  aria-describedby="tooltip-550000"
                  class="usa-tooltip__trigger"
                  data-position="top"
                  tabindex="0"
                >
                  Top Sources/Media
                </span>
                <span
                  aria-hidden="true"
                  class="usa-tooltip__body"
                  id="tooltip-550000"
                  role="tooltip"
                >
                  A source is a representation of the publisher or inventory source from which traffic originated. A medium is a method for acquiring users to websites or applications.
                </span>
              </span>
            </a>
          </div>
          <div
            class="card:grid-col-12 mobile-lg:grid-col-auto"
          >
            <div
              class="display-flex card:flex-justify-center mobile-lg:flex-justify-end"
            >
              <select
                aria-label="session source medium chart time filter"
                class="usa-select chart__filter margin-top-0"
              >
                <option
                  value="top-session-source-medium-yesterday"
                >
                  Yesterday
                </option>
                <option
                  value="top-session-source-medium-7-days"
                >
                  7 Days
                </option>
                <option
                  value="top-session-source-medium-30-days"
                >
                  30 Days
                </option>
                <option
                  value="top-session-source-medium-90-days"
                >
                  90 Days
                </option>
                <option
                  value="top-session-source-medium-current-year"
                >
                  Current Year
                </option>
                <option
                  value="top-session-source-medium-current-fiscal-year"
                >
                  Current Fiscal Year
                </option>
                <option
                  value="top-session-source-medium-previous-year"
                >
                  Previous Year
                </option>
                <option
                  value="top-session-source-medium-previous-fiscal-year"
                >
                  Previous Fiscal Year
                </option>
              </select>
            </div>
          </div>
        </div>
        <div
          class="text--capitalize"
        >
          <figure
            id="chart_session_source_medium"
          >
            <div
              class="data chart__bar-chart margin-top-2"
            />
          </figure>
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
